app/controllers/distributions_controller.rb

```diff
@@ -164,4 +165,17 @@ def filter_params
 
     params.require(:filters).slice(:by_item_id, :by_partner)
   end
+
+  def find_distribution
+    Distribution.includes(:line_items).includes(:storage_location).find(params[:id])
```

Comment:
This should probably be memoized, given how it's used:

```suggestion
    @distribution ||= Distribution.includes(:line_items).includes(:storage_location).find(params[:id])
```

You'll need to change the method name to `distribution` (or Rubocop will complain). The lines that say `@distribution = find_distribution` can be also removed if you add a `before_action` like you did with `:allow_edit?`, constraining it to the list of actions above that need it.

(I don't know if you're familiar with memoization or not -- if you aren't, [here's an article](https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/))

`allow_edit?` can have line 174 struck as well, should you do this.

Reply:
I agree with you! I really don't think about memoization...